File: packages/contract-py/src/tigr_asgi_contract/ids.py
from pydantic import BaseModel


class UnitIds(BaseModel):
    unit_id: str | None = None
    parent_unit_id: str | None = None
    session_id: str | None = None
    stream_id: int | None = None
    datagram_id: int | None = None
    emit_id: str | None = None

File: packages/contract-rs/src/ids.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct UnitIds {
    pub unit_id: Option<String>,
    pub parent_unit_id: Option<String>,
    pub session_id: Option<String>,
    pub stream_id: Option<u64>,
    pub datagram_id: Option<u64>,
    pub emit_id: Option<String>,
}
